frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1146,13 +1146,15 @@ function ReaderHighlight:getHighlightBookmarkItem()
         local datetime = os.date("%Y-%m-%d %H:%M:%S")
         local page = self.ui.document.info.has_pages and
                 self.hold_pos.page or self.selected_text.pos0
+        local chapter_name = self.ui.toc:getAccurateTocTitleByXPointer(self.selected_text.pos0)
```

Comment:
I think you want chapters working too with PDF, that don't use xpointers. For PDF, we can't do better than using the getTocTitleByPage you previously used.
May be hide that business in betterly named `ReaderToc:getTocChapter(pn_or_xp)`

Reply:
I moved `getAccurateTocTitleByXPointer` to `getTocTitleByPage`, it uses the xpointer for accuracy(xpointer is passed), otherwise it uses page, as it used to.